lib/cfg.c

```diff
@@ -365,7 +365,11 @@ cfg_init(GlobalConfig *cfg)
   log_template_options_init(&cfg->template_options, cfg);
   if (!cfg_init_modules(cfg))
     return FALSE;
-  return cfg_tree_start(&cfg->tree);
+  if (!cfg_tree_start(&cfg->tree))
+    return FALSE;
+
+  g_assert(cfg_tree_on_inited(&cfg->tree));
```

Comment:
asserting on the return value: does it mean that `cfg_tree_on_inited` should not have return value?
`log_threaded_source_driver_start_worker` returns `TRUE` unconditionally too. Just an idea, but we might want to turn every on_inited to void consistently?

Reply:
Since I think someone may want syslog-ng to be terminated if there is an error when `cfg_tree_on_inited`, I use `g_assert` here to achieve this goal.